### ISSUE TITLE
Problems to recover error messages after switching to the Dio package.

### ISSUE THREAD

**Issue by RodrigoSMarques** [parse-community/Parse-SDK-Flutter#481]
I identified two problems in version 1.0.28 that do not happen in 1.0.27.

The problem is related to switching to the Dio package.

1) When executing a function, I use try / catch to retrieve the error message. The error message (String) is no longer being returned. A DioError is being returned.

Version 1.0.27 - expected response
```dart
flutter: ╭-- Parse Request
flutter: curl -X POST -H 'content-type: text/plain; charset=utf-8' -H 'user-agent: Flutter Parse SDK 1.0.27' -H 'X-Parse-Application-Id: xxxxxxxxxxx' -H 'X-Parse-Session-Token: r:xxxxxxxxxxx' -H 'X-Parse-Client-Key: xxxxxxxxxxx' -d '{"code":"TESTE10","minOrderValue":100.0,"discountValue":10.0,"qtyTotal":9999,"expirationDate":1602817140000,"advertiserID":"iUAzCRbUcQ"}' https://xxx.xxxxxxx.com/1/functions/saveCoupon

 https://xxx.xxxxxxx.com/1/functions/saveCoupon
flutter: ╰--
flutter: ╭-- Parse Response
Class: saveCoupon
Function: ParseApiRQ.execute
Status Code: 141
Type: ScriptError
Error: Coupon Code in use. Try another code
╰--
```
Version 1.0.28 - error
```dart
flutter: ╭-- Parse Request
flutter: curl -X POST -H 'content-type: text/plain; charset=utf-8' -H 'user-agent: Flutter Parse SDK 1.0.28' -H 'X-Parse-Application-Id: xxxxxxxxxx' -H 'X-Parse-Session-Token: r:xxxxxxxxxxx' -H 'X-Parse-Client-Key: xxxxxxxxxxx' -d '{"code":"TESTE10","minOrderValue":100.0,"discountValue":10.0,"qtyTotal":9999,"expirationDate":1602817140000,"advertiserID":"iUAzCRbUcQ"}' https://xxx.xxxxxxx.com/1/functions/saveCoupon
 https://xxx.xxxxxxx.com/1/functions/saveCoupon
flutter: ╰--
[VERBOSE-2:ui_dart_state.cc(177)] Unhandled Exception: type 'DioError' is not a subtype of type 'String'
```

2) When an error occurs when saving an object and an exception is generated, the error is not being returned.

Version 1.0.27 - expected response
```dart
flutter: ╭-- Parse Request
flutter: curl -X POST -H 'content-type: text/plain; charset=utf-8' -H 'user-agent: Flutter Parse SDK 1.0.27' -H 'X-Parse-Application-Id: xxxxxxxxxx' -H 'X-Parse-Session-Token: r:xxxxxxxxxxx' -H 'X-Parse-Client-Key: xxxxxxxxxxx' -d '{"ACL":{"*":{"read":true}},"advertiser":{"__type":"Pointer","className":"tbAdvertiser","objectId":"zQYUoBZpP3"},"customerId":"q3ssdPfvUC"<…>
flutter: ╰--
flutter: ╭-- Parse Response
Class: tbOrder
Function: ParseApiRQ.create
Status Code: 111
Type: IncorrectType
Error: schema mismatch for order.status; expected Number but got String
╰--
```
Version 1.0.28 - error
```dart
flutter: ╭-- Parse Request
flutter: curl -X POST -H 'content-type: null' -H 'user-agent: Flutter Parse SDK 1.0.28' -H 'X-Parse-Application-Id: xxxxxxxxxx' -H 'X-Parse-Session-Token: r:xxxxxxxxxxx' -H 'X-Parse-Client-Key: xxxxxxxxxxx' https://xxx.xxxxxxx.com/1/classes/order

 https://xxx.xxxxxxx.com/1/classes/order
flutter: ╰--
flutter: ╭-- Parse Response
Class: order
Function: ParseApiRQ.create
Status Code: -1
Type: UnknownError
Exception: DioError [DioErrorType.RESPONSE]: Http status error [400]
╰--
```

**Comment by fischerscode** [comment]
@RodrigoSMarques 
I think I found a fix.
Could you please give [this branch](https://github.com/fischerscode/Parse-SDK-Flutter/tree/dio-error-response) a try?

**Comment by fischerscode** [comment]
@RodrigoSMarques 
After cloning, you have to change `parse_server_sdk: ^2.0.0` to
```yaml
  parse_server_sdk:
    path: ../dart
```
in packages/flutter/pubspec.yaml

**Comment by RodrigoSMarques** [comment]
@fischerscode 
I ran the tests and it is working as in version 1.0.27

A release with the fix in version 2.0.1 would be interesting.

What do you think about this?

**Comment by fischerscode** [comment]
> I ran the tests and it is working as in version 1.0.27

Great!
> A release with the fix in version 2.0.1 would be interesting.

I agree that this is an important fix and releasing 2.0.1 might be a good idea. https://github.com/parse-community/Parse-SDK-Flutter/pull/482#issuecomment-708297643
